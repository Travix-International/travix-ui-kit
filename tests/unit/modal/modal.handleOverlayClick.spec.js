import React from 'react';
import { shallow } from 'enzyme';
import Modal from '../../../components/modal/modal';

describe('Modal: handleOverlayClick', () => {
  it('should close on click overlay', () => {
    Modal.prototype.close = jest.fn();

    const component = shallow(
      <Modal
        active
      >
        Modal Content
      </Modal>
    );
    expect(Modal.prototype.close).not.toBeCalled();

    component.instance().handleOverlayClick();
    expect(Modal.prototype.close).toBeCalled();
  });

  it('should call "onOverlayClick" prop after click on overlay', () => {
    const onOverlayClick = jest.fn();

    const component = shallow(
      <Modal
        active
        closeOnOverlayClick={false}
        onOverlayClick={onOverlayClick}
      >
        Modal Content
      </Modal>
    );
    expect(onOverlayClick).not.toBeCalled();

    component.instance().handleOverlayClick();
    expect(onOverlayClick).toBeCalled();
    expect(component).toMatchSnapshot();
  });
});
