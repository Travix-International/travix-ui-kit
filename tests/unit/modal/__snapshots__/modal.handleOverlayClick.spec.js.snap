// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal: handleOverlayClick should call "onOverlayClick" prop after click on overlay 1`] = `
<Global
  className="ui-modal ui-modal_active"
  noscroll={false}
>
  <div
    className="ui-modal__overlay"
    onClick={[Function]}
  />
  <div
    className="ui-modal__container"
  >
    <section
      className="ui-modal__content"
    >
      Modal Content
    </section>
  </div>
</Global>
`;
